jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/rsu/RsuDepositor.java

```diff
@@ -127,9 +127,10 @@ public void run() {
 							logger.error("Error on RSU SNMP deposit to {}: message already exists at index {}.", curRsu.getRsuTarget(),
 									destIndex);
 						} else {
-							// Misc error
-							logger.error("Error on RSU SNMP deposit to {}: {}", curRsu.getRsuTarget(), "Error code "
-									+ rsuResponse.getResponse().getErrorStatus() + " " + rsuResponse.getResponse().getErrorStatusText());
+							// Misc error, provide context
+							logger.error("Error on RSU SNMP deposit to {}: {}", curRsu.getRsuTarget(), "Error code '"
+									+ rsuResponse.getResponse().getErrorStatus() + "' '" + rsuResponse.getResponse().getErrorStatusText()
+									+ "' Request PDU: " + rsuResponse.getRequest() + " Response PDU: " + rsuResponse.getResponse());
```

Comment:
Is it worth logging both the Request and Response PDUs? I'm assuming this is to check for type mismatch on the request/response PDUs but I could see this being confusing for someone who isn't familiar with these objects. Would it be worthwhile to create a method to extract the type of a PDU and log that instead of the entire PDU?

Reply:
I think it is worthwhile to include these PDUs when logging a miscellaneous SNMP error. A likely first question when encountering these errors is "What did the PDUs look like?" and checking for snmp protocol mismatch may not always be sufficient given the variety of error statuses that can be returned.

Reply:
I decided to split out the PDU logging into a separate debug statement, rather than tacking it onto the end of the error statement.